crates/swc_isograph_plugin/src/lib.rs

```diff
@@ -0,0 +1,463 @@
+
+    let root_dir = config.root_dir;
+
+    let isograph_config = config.isograph_project_config;
+
+    debug!("Config: {:?}", isograph_config);
+    let file_name = metadata
+        .get_context(&TransformPluginMetadataContextKind::Filename)
+        .unwrap_or_default();
+    let path = Path::new(&file_name);
+
+    let mut isograph = compile_iso_literal_visitor(
+        &isograph_config,
+        path,
+        root_dir.as_path(),
+        Some(metadata.unresolved_mark),
+    );
+
+    program.fold_with(&mut isograph)
+}
+
+pub fn compile_iso_literal_visitor<'a>(
+    config: &'a IsographProjectConfig,
+    filepath: &'a Path,
+    root_dir: &'a Path,
+    unresolved_mark: Option<Mark>,
+) -> impl Fold + 'a {
+    IsoLiteralCompilerVisitor {
+        config,
+        filepath,
+        unresolved_mark,
+        imports: vec![],
+        root_dir,
+    }
+}
+
+#[derive(Error, Clone, Debug, Eq, PartialEq)]
+enum IsographTransformError {
+    #[error("Invalid iso tag usage. Expected 'entrypoint' or 'field'.")]
+    InvalidIsoKeyword,
+
+    #[error("Invalid iso tag usage. The iso function should be passed exactly one argument.")]
+    IsoFnCallRequiresOneArg,
+
+    #[error("Iso invocation require one parameter.")]
+    IsoRequiresOneArg,
+
+    #[error("Malformed iso literal. I hope the iso compiler failed to accept this literal!")]
+    MalformedIsoLiteral,
+
+    #[error("Only template literals are allowed in iso fragments.")]
+    OnlyAllowedTemplateLiteral,
+
+    #[error("Substitutions are not allowed in iso fragments.")]
+    SubstitutionsNotAllowedInIsoFragments,
+}
+
+fn show_error(span: &Span, err: &IsographTransformError) -> Result<(), anyhow::Error> {
+    let msg = IsographTransformError::to_string(err);
+
+    HANDLER.with(|handler| {
+        handler.struct_span_err(*span, &msg).emit();
+    });
+    bail!(msg)
+}
+
+#[derive(Debug, Clone)]
+struct IsographImport {
+    path: Atom,
+    item: Atom,
+    unresolved_mark: Option<Mark>,
+}
+
+impl IsographImport {
+    fn as_module_item(&self) -> ModuleItem {
+        ModuleItem::ModuleDecl(ModuleDecl::Import(ImportDecl {
+            span: Default::default(),
+            specifiers: vec![ImportSpecifier::Default(ImportDefaultSpecifier {
+                span: Default::default(),
+                local: Ident {
+                    span: self
+                        .unresolved_mark
+                        .map(|m| DUMMY_SP.apply_mark(m))
+                        .unwrap_or_default(),
+                    sym: self.item.clone(),
+                    optional: false,
+                },
+            })],
+            src: Box::new(self.path.clone().into()),
+            type_only: false,
+            with: None,
+            phase: Default::default(),
+        }))
+    }
+}
+
+#[derive(Deserialize, Default, Debug, Clone, Copy, PartialEq)]
+pub enum ArtifactType {
+    #[default]
+    Entrypoint,
+    Field,
+    Unknown,
```

Comment:
Let's remove the `Unknown` variant, we don't ever produce it.

It also doesn't need to be pub, since it's not exposed. (I would expect that, if it's not pub, the compiler would warn that `Unknown` is never created. It doesn't. Odd!)